org.eclipse.gef/src/org/eclipse/gef/ui/palette/FlyoutPaletteComposite.java

```diff
@@ -1557,37 +1559,42 @@ private static class DragCursors {
 		 * @return the cursor
 		 */
 		public static Cursor getCursor(int code) {
-			Display display = Display.getCurrent();
 			if (cursors[code] == null) {
-				ImageDescriptor source = null;
-				ImageDescriptor mask = null;
 				switch (code) {
 				case LEFT:
-					source = PlatformUI.getWorkbench().getSharedImages()
-							.getImageDescriptor(ISharedImages.IMG_OBJS_DND_LEFT_SOURCE);
-					mask = PlatformUI.getWorkbench().getSharedImages()
-							.getImageDescriptor(ISharedImages.IMG_OBJS_DND_LEFT_MASK);
-					cursors[LEFT] = new Cursor(display, source.getImageData(), mask.getImageData(), 16, 16);
+					cursors[LEFT] = createCursor(ISharedImages.IMG_OBJS_DND_LEFT_SOURCE,
+							ISharedImages.IMG_OBJS_DND_LEFT_MASK);
 					break;
 				case RIGHT:
-					source = PlatformUI.getWorkbench().getSharedImages()
-							.getImageDescriptor(ISharedImages.IMG_OBJS_DND_RIGHT_SOURCE);
-					mask = PlatformUI.getWorkbench().getSharedImages()
-							.getImageDescriptor(ISharedImages.IMG_OBJS_DND_RIGHT_MASK);
-					cursors[RIGHT] = new Cursor(display, source.getImageData(), mask.getImageData(), 16, 16);
+					cursors[RIGHT] = createCursor(ISharedImages.IMG_OBJS_DND_RIGHT_SOURCE,
+							ISharedImages.IMG_OBJS_DND_RIGHT_MASK);
 					break;
 				default:
 				case INVALID:
-					source = PlatformUI.getWorkbench().getSharedImages()
-							.getImageDescriptor(ISharedImages.IMG_OBJS_DND_INVALID_SOURCE);
-					mask = PlatformUI.getWorkbench().getSharedImages()
-							.getImageDescriptor(ISharedImages.IMG_OBJS_DND_INVALID_MASK);
-					cursors[INVALID] = new Cursor(display, source.getImageData(), mask.getImageData(), 16, 16);
+					cursors[INVALID] = createCursor(ISharedImages.IMG_OBJS_DND_INVALID_SOURCE,
+							ISharedImages.IMG_OBJS_DND_INVALID_MASK);
 					break;
 				}
 			}
 			return cursors[code];
 		}
 
+		/**
+		 * Creates a new cursor using the shared images as source and mask,
+		 * respectively. The cursors are created with respect to the current device
+		 * zoom, with the hotspot being the center of the image.
+		 */
```

Comment:
Add the parameters to the javadoc and mention that they need to be names of shared cursors.